pyro/distributions/zero_inflated.py

```diff
@@ -70,17 +71,29 @@ def log_prob(self, value):
         if self._validate_args:
             self._validate_sample(value)
 
+        support = self.support
+        epsilon = abs(torch.finfo(value.dtype).eps)
+        zero_idx = (value == 0)
+
+        if hasattr(support, "lower_bound"):
+            if support.lower_bound == 0.0:
+                value = value.clamp_min(epsilon)
+
+        if hasattr(support, "upper_bound"):
+            if support.upper_bound == 1.0:
+                value = value.clamp_max(1 - epsilon)
+
         if "gate" in self.__dict__:
             gate, value = broadcast_all(self.gate, value)
             log_prob = (-gate).log1p() + self.base_dist.log_prob(value)
-            log_prob = torch.where(value == 0, (gate + log_prob.exp()).log(), log_prob)
+            log_prob = torch.where(zero_idx == True, (gate + log_prob.exp()).log(), log_prob)
```

Comment:
nit: simplify `zero_idx == True` to `zero_idx`

Reply:
Yes good one, I change it from `zero_idx == True` to `zero_idx`.

Actually, the reason why I changed the import is that my IDE (Dataspell) indicated that it cannot import the `NegativeBinomial` from pyro. But I just tried it in the console and it is working, so I move back to the initial imports.

Thanks for pointing me to these issues.